crates/polars-core/src/chunked_array/mod.rs

```diff
@@ -75,8 +75,8 @@ pub type ChunkLenIter<'a> = std::iter::Map<std::slice::Iter<'a, ArrayRef>, fn(&A
 ///
 /// ```rust
 /// # use polars_core::prelude::*;
-/// fn apply_cosine_and_cast(ca: &Float32Chunked) -> Float64Chunked {
-///     ca.apply_values_generic(|v| v.cos() as f64)
+/// fn apply_cosine_and_cast(ca: &Float32Chunked) -> Float32Chunked {
+///     ca.apply(|v| v.cos())
```

Comment:
the description above says

> This allows us to apply closures to the data and collect the results to a [`ChunkedArray`] of the same type `T`.

so I think a plain `apply` might be a better example here?